### PR TITLE
Controller logic bll

### PR DESCRIPTION
**Pull request by oleespen**

### CODE REVIEW

File: src/GroupProject/BLL/UserBLL.cs [hunk 1]
```diff
@@ -49,6 +80,36 @@ internal bool deleteInvoice(ApplicationUser user, int id)
...
+            return getAccounts(user).Where(item => item.kontoType != Konto.kontoNavn.BSU);
+        }
+
+        public Konto checkAccount(ApplicationUser user, PaymentViewModel model)
+        {
+            return getAccounts(user).Find(acc => acc.kontoNr == model.fromAccount);
+        }
+
+        public PaymentViewModel changeInvoice(Betalinger invoice)
+        {
```
Comment (ArgTang) [inline src/GroupProject/BLL/UserBLL.cs:91]: maybe convertToViewModel?

File: src/GroupProject/BLL/UserBLL.cs [hunk 2]
```diff
@@ -24,22 +28,49 @@ public List<Konto> getAccounts(ApplicationUser user)
...
-            return _dbAccess.getPayments(user);
-            
+            List<Betalinger> list = _dbAccess.getPayments(user);
+            list.Sort((x, y) => x.forfallDato.CompareTo(y.forfallDato));
+            return list;
         }
 
-        public void changePayment(Betalinger betal)
+        public void changePayment(PaymentViewModel model, Betalinger betaling, Konto account, ApplicationUser user)
         {
```
Comment (ArgTang) [inline src/GroupProject/BLL/UserBLL.cs:39]: Maybe call the functions Update?

File: src/GroupProject/BLL/UserBLL.cs [hunk 3]
```diff
@@ -24,22 +28,49 @@ public List<Konto> getAccounts(ApplicationUser user)
...
+            betaling.konto = account;
+            betaling.tilKonto = model.toAccount;
+            betaling.belop = new Decimal(Double.Parse(model.amount + "," + model.fraction));
+            betaling.info = model.paymentMessage;
+            betaling.utfort = false;
+            betaling.kid = model.kid;
+            betaling.mottaker = model.reciever;
+            betaling.forfallDato = model.date;
+            betaling.UpdatedDate = DateTime.Now;
+            betaling.UpdatedBy = user.UserName;
```
Comment (ArgTang) [inline src/GroupProject/BLL/UserBLL.cs]: We do this conversion two places, maybe make a privae function to do this?

**Review by ArgTang — DISMISSED** [review]

File: src/GroupProject/BLL/UserBLL.cs (hunk 2 above)
Comment (oleespen) [reply src/GroupProject/BLL/UserBLL.cs:39]: done

File: src/GroupProject/BLL/UserBLL.cs (hunk 3 above)
Comment (oleespen) [reply src/GroupProject/BLL/UserBLL.cs]: done

File: src/GroupProject/BLL/UserBLL.cs (hunk 1 above)
Comment (oleespen) [reply src/GroupProject/BLL/UserBLL.cs:91]: done

**Review by aleksanderaleksic — APPROVED** [review]

**Pull request merged**